### PR TITLE
[Props] Add dimension property

### PR DESCRIPTION
**Pull request by zhoonit**
- [Props] Add dimension property

```
This patch add dimension property

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

File: nntrainer/utils/base_properties.cpp [hunk 1]
```diff
@@ -48,4 +50,40 @@ unsigned int str_converter<uint_prop_tag, unsigned int>::from_string(
...
+    << "More than 4 axes is not supported, target string: " << value;
+
+  TensorDim target;
+
+  int cur_axis = 3;
+  for (auto iter = tokens.rbegin(); iter != tokens.rend(); iter++) {
+    target.setTensorDim(cur_axis--, std::stoi(*iter));
+  }
+
+  std::cerr << target;
```
Comment (kparichay) [inline nntrainer/utils/base_properties.cpp:84]: is this intended?

File: nntrainer/utils/base_properties.cpp [hunk 2]
```diff
@@ -48,4 +50,40 @@ unsigned int str_converter<uint_prop_tag, unsigned int>::from_string(
...
+  }
+
+  NNTR_THROW_IF(tokens.size() > MAXDIM, std::invalid_argument)
+    << "More than 4 axes is not supported, target string: " << value;
+
+  TensorDim target;
+
+  int cur_axis = 3;
+  for (auto iter = tokens.rbegin(); iter != tokens.rend(); iter++) {
+    target.setTensorDim(cur_axis--, std::stoi(*iter));
```
Comment (kparichay) [inline nntrainer/utils/base_properties.cpp:81]: stoi -> stoul?

File: test/unittest/unittest_properties.cpp [hunk 3]
```diff
@@ -111,6 +127,27 @@ TEST(BasicProperty, valid_p) {
...
+    EXPECT_EQ(q.get(), nntrainer::TensorDim(1, 2, 3, 4));
+    EXPECT_EQ(nntrainer::to_string(q), "1:2:3:4");
+  }
+
+  { /**< from_string -> get / to_string, dimension */
+    DimensionOfBanana q;
+    nntrainer::from_string("3:4", q);
+    EXPECT_EQ(q.get(), nntrainer::TensorDim(1, 1, 3, 4));
+    EXPECT_EQ(nntrainer::to_string(q), "1:1:3:4");
+  }
```
Comment (kparichay) [inline test/unittest/unittest_properties.cpp:149]: can you please a couple more unittests (negative ones) - one with empty string, only ":" and more than 4 dimensions.

File: nntrainer/utils/base_properties.cpp (hunk 1 above)
Comment (zhoonit) [reply nntrainer/utils/base_properties.cpp:84]: my bad. Thanks :)

File: nntrainer/utils/base_properties.cpp (hunk 2 above)
Comment (zhoonit) [reply nntrainer/utils/base_properties.cpp:81]: Updated

File: test/unittest/unittest_properties.cpp (hunk 3 above)
Comment (kparichay) [reply test/unittest/unittest_properties.cpp:149]: i guess more than 4 dimensions is already added here - https://github.com/nnstreamer/nntrainer/pull/1161/files#diff-8bd026869e0f1961b1ba9f517c2af8ef4be981ebfd65959d1410a355d5829a9bR225

Comment (zhoonit) [reply test/unittest/unittest_properties.cpp:149]: Done.

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by kparichay — APPROVED** [review]
LGTM

**Review by lhs8928 — APPROVED** [review]
LGTM

**Pull request merged**